core/iwasm/interpreter/wasm_loader.c

```diff
@@ -7648,6 +7751,24 @@ wasm_loader_prepare_bytecode(WASMModule *module, WASMFunction *func,
                     goto fail;
 
 #if WASM_ENABLE_DYNAMIC_PGO != 0
+                struct WASMProfCntInfo *opcode_cnt;
+
+                opcode_cnt = loader_malloc(sizeof(struct WASMProfCntInfo),
+                                           error_buf, error_buf_size);
+                if (!opcode_cnt)
+                    goto fail;
+
+                opcode_cnt->func_idx =
+                    module->import_function_count + cur_func_idx;
+                opcode_cnt->offset = p - func->code;
+                opcode_cnt->opcode = WASM_OP_BR_IF;
+                opcode_cnt->counter_amount = 2;
+                opcode_cnt->first_counter_idx = ent_and_br_cnts_cap;
+
+                if (bh_list_append(func_prof_cnts_info, opcode_cnt)
+                    != BH_LIST_SUCCESS)
+                    goto fail;
```

Comment:
Same as above

Reply:
I am hoping `wasm_loader_unload()` (triggered by `goto fail` -> `return false` of `load_from_sections()`) will free
allocated resources.